roles/api.py

```diff
@@ -162,3 +164,35 @@ def remove_user_from_repo_group(
         repo_group = Group.objects.get(name=group_type.format(repo.slug))
         user.groups.remove(repo_group)
         user.save()
+
+
+def list_users_in_repo(repo, base_group_type=None):
+    """
+    Lists all the users in the repository groups
+    If the group type is specified, the list is limited to that group
+
+    Args:
+        repo (learningresources.models.Repository): repository used to extract
+            the right group to use
+        base_group_type (roles.permissions.BaseGroupTypes): group string
+    Returns:
+        list (list of User): list of users in one or all the repository groups
+    """
+    users_groups = []
+    if base_group_type is not None:
+        if not BaseGroupTypes.is_base_group_type(base_group_type):
+            raise InvalidGroupType
+        base_group_types = [base_group_type]
+    else:
+        base_group_types = BaseGroupTypes.all_base_groups()
+    with transaction.atomic():
+        for base_group_type in base_group_types:
+            group = Group.objects.get(
+                name=GroupTypes.get_repo_groupname_by_base(
+                    base_group_type
+                ).format(repo.slug)
+            )
+            users_groups = users_groups + [
```

Comment:
Can you use `+=` here? It looks cleaner to me and may be slightly more efficient but feel free to leave it if you disagree